sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -961,7 +961,11 @@ private[hive] object HiveClientImpl {
   /** Get the Spark SQL native DataType from Hive's FieldSchema. */
   private def getSparkSQLDataType(hc: FieldSchema): DataType = {
     try {
-      CatalystSqlParser.parseDataType(hc.getType)
+      hc.getType match {
+        // SPARK-28313 compatible hive void type
+        case "void" => NullType
```

Comment:
I believe this is the fix we need. But I'm curious about when can this happen if hive forbids defining void type columns.

Reply:
Fix this point,  hive void type will be converted to null type. 
Such as, t1 have a void type column c1, when execute `show create table $tbl`.
Spark:
`create table t1 (c1 null)`
Hive:
`create table t1 (c1 void)`

Reply:
> But I'm curious about when can this happen if hive forbids defining void type columns.

Can hive do `create table t1 (c1 void)`?

Reply:
No, hive can not but hive can do `create table t1 as select null as c1`.

Reply:
can we make this fix surgical and only keep this change?

Reply:
How about add code to check and change null to void at `ShowCreateTableCommand.showCreateHiveTable()` ?

Reply:
Does it have to be in this PR? If it has to, I'm ok with it.

Reply:
Yeah~ If you have a time, you could review the latest commit.